crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -961,10 +962,30 @@ def upgrade_cluster(founder, cluster)
         Rails.logger.info("Nodes in cluster #{cluster} successfully upgraded")
       end
 
+      # Rabbitmq uses native clustering in SOC8, we need to adapt the configuration
+      def upgrade_rabbitmq_setup
+        rabbit_cluster_members = ::Node.find(
+          "run_list_map:pacemaker-cluster-member AND run_list_map:rabbitmq-server "
+        )
+        return if rabbit_cluster_members.empty?
+
+        prop = Proposal.find_by(barclamp: "rabbitmq", name: "default")
+        unless prop.raw_data["attributes"]["rabbitmq"]["cluster"]
+          prop.raw_data["attributes"]["rabbitmq"]["cluster"] = true
+          prop.save
+        end
+
+        rabbit_cluster_members.each do |node|
+          node_api = Api::Node.new node.name
```

Comment:
Arent they already `Node` objects here? Do we really need to transform them into an `Api::Node` object to save 2 variables?

Im just saying, having the node attribute modification here instead that on the `Api::Node` part, makes it clearer what this is doing and we can add comments to make clear that we need the same attrs on the Proposal and on the Node.

Reply:
Yeah. there's extra search, which seems unnecessary

So you think I should move `update_rabbitmq_config` to `Node`, or are you fine with saving it directly from here?